### PR TITLE
feat(cli): cache connection inside client

### PR DESCRIPTION
**Pull request by sundy-li**

### CODE REVIEW

**Comment by youngsofun** [comment]
mainly for flight SQL?
what is the benefit to cache a restful client?

**Comment by sundy-li** [comment]
> mainly for flight SQL? what is the benefit to cache a restful client?

Wo cache the HTTP connection pool inside hyper.

**Comment by everpcpc** [comment]
I do not understand why we are doing this. Storing connection with session state may get unexpected behavior.
For example, I get connection 1, set a config, and put it back. Then I get a connection 2, if the previous connection is reused, I would get a connection with unexpected session config different from which in the DSN.

**Comment by everpcpc** [comment]
If the method name `get_conn` is not clear, we could rename it to `new_conn` or something else.

**Comment by sundy-li** [comment]
Currently,  we are not reusing `session state `.

We just reuse the connection in Hyper's connection pool.

For example, the `get_conn`  function will initialize several idle HTTP connections inside the pool (with keep-alive).

This pr only allows us to fetch idle connection from initialized client.

File: core/src/client.rs [hunk 1]
```diff
@@ -82,6 +82,12 @@ impl APIClient {
         Ok(client)
     }
 
+    pub fn reuse(&self) -> Self {
+        let mut client = self.clone();
+        client.session_state = Arc::new(Mutex::new(SessionState::default()));
```
Comment (sundy-li) [inline core/src/client.rs:87]: session state is set to default in resue function.

**Pull request closed without merging**